volume/rsync/rsync.go

```diff
@@ -429,18 +428,18 @@ func (v *RsyncVolume) Snapshot(label, message string, tags []string) (err error)
 		}
 		return err
 	}
-
-	//make sure none of the tags already exist
+	// check the tags for duplicates
 	for _, tagName := range tags {
-		if info, err := v.GetSnapshotWithTag(tagName); err != nil {
-			glog.Errorf("Error checking for tag conflicts: %v", err)
-			return err
-		} else if info != nil {
-			glog.Errorf("Can not create snapshot %s, tag '%s' already exists", label, tagName)
-			return volume.ErrTagAlreadyExists
+		if tagInfo, err := v.GetSnapshotWithTag(tagName); err != volume.ErrSnapshotDoesNotExist {
+			if err != nil {
+				glog.Errorf("Could not look up snapshot for tag %s: %s", tagName, err)
+				return err
+			} else {
```

Comment:
Just a nit, but I'd rather be explicit and check the value of `tagInfo` instead of inferring that it's not `nil` through the value of `err`.